W-22433617 Update flex-gateway-llm-proxy.adoc (#1059)
mulesoft/docs-gateway: gateway/1.12/modules/ROOT/pages/flex-gateway-llm-proxy-create-llm-proxy.adoc

[[configure-semantic-routing]]
== Configure Semantic Routing

To configure semantic routing:

> . Ensure you have already xref:flex-gateway-llm-proxy-semantic-service.adoc[Configured a semantic service].

Reviewer: ```suggestion
. Make sure you have already xref:flex-gateway-llm-proxy-semantic-service.adoc[Configured a semantic service].
```

. Configure multiple routes and select a target model for each route. Click *Add LLM Route* to create new routes.
. Select *Semantic* for *Routing strategy*.
. Click *Select a service* and select a service.

> . Define or select a prompt topics for the routes:

Reviewer: ```suggestion
. Define or select a prompt topic for the routes:
```

** Advanced scale semantic service:
... Select prompt topics from your predefined prompt topics.
** Basic scale semantic service:
... Click the *Select prompt topics*.
... Click *+ Create prompt topic*.
... Define a *Prompt topic name*.
... Define a *Prompt utterances* or click *Upload utterances* to upload a plain text file containing your prompt utterances.
... Click *Create*.
... Create multiple prompt topics for each route as needed.
. Configure a *Fallback route* for the request to be sent to if it doesn't match a semantic route:
.. Specify an accuracy threshold. When the accuracy of the semantic match is less than this threshold, traffic is sent to the fallback route.
.. Select a *Route* to fallback to.
.. Select a *Target model* for the fallback route to use.
. Create a *Semantic prompt guard* to block users from asking the server about specific topics:
** Advanced scale semantic service:
... Select topics from your predefined prompt topics.
** Basic scale semantic service:
... Click *+ Create deny list*.
... Define a *Prompt topic name*.

> ... Define a *Prompt utterances* or click *Upload utterances* to upload a plain text file containing your prompt utterances.

Reviewer: ```suggestion
... Define prompt utterances or click *Upload utterances* to upload a plain text file containing your prompt utterances.
```
